src/NewsAggregator.php

```diff
@@ -1,46 +1,35 @@
 <?php
-
 namespace App;
 
-use Package\FoxNews\FoxNews;
-use Package\NYTimes\NewYorkTimes;
+use App\Factories\NewsAggregatorFactory;
+use Exception;
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
 
-class NewsAggregator
+final class NewsAggregator
 {
-    private $foxNews;
-    private $newYorkTimes;
 
-    public function __construct()
-    {
-        $this->foxNews = new FoxNews();
-        $this->newYorkTimes = new NewYorkTimes();
-    }
+    /** @var array */
+    private array $news = [];
 
-    public function get()
+    public function __construct ()
     {
-        $news = [];
-        foreach ($this->foxNews->getNewsFromAPI()['articles'] as $row) {
-            $news[] = [
-                'title'        => $row['title'],
-                'author'       => $row['author'],
-                'image'        => $row['urlToImage'],
-                'publish_date' => $row['publishedAt'],
-                'source'       => $row['source']['name'],
-                'url'          => $row['url'],
-            ];
-        }
+        foreach (NewsAggregatorFactory::newsAggregatorsList() as $key => $newsAggregator) {
```

Comment:
I think it makes more sense to move this code to the `get()` method. Is it necessary to have it in the constructor?

Reply:
no, it is not necessary to have it in the constructor